Q66: Plus One

```diff
@@ -0,0 +1,62 @@
+/*
+    Runtime: 1 ms, faster than 10.74% of Java online submissions for Plus One.
+    Memory Usage: 37.8 MB, less than 72.32% of Java online submissions for Plus One.
+    
+    Difficulty: 5
+    
+    Time: 42 minutes and 23 seconds
+    
+    It took some time to add one to the end of the number because of the corner cases where I have to carry the digits.
+    There was a lot of booleans to use in order to keep track of whether I needed to add or to carry.
+*/
+
+class Solution {
+    public int[] plusOne(int[] digits) {
+        //Creates an array list to keep a running count of the digits
+        ArrayList<Integer> al = new ArrayList<Integer>();
+        //Boolean to see if we need to carry
+        boolean carry = false;
+        //Boolean to see if we have already added
+        boolean added = false;
+        //For every digit starting from the ones digit, add one and add it to the arraylist. 
+        //If there is a carry keep on carrying until it stops.
+        for(int digit = digits.length-1; digit >= 0; digit--){
```

Comment:
Since the intention of the loop is to break as early as possible, the `for-loop` may not be the best for reading. Whenever "break" is the core of algorithm, use `while`-loop.